lightning/src/ln/channelmanager.rs

```diff
@@ -8914,15 +8922,19 @@ macro_rules! create_refund_builder { ($self: ident, $builder: ty) => {
 		let nonce = Nonce::from_entropy_source(entropy);
 		let context = OffersContext::OutboundPayment { payment_id, nonce, hmac: None };
 
-		let path = $self.create_blinded_paths_using_absolute_expiry(context, Some(absolute_expiry))
-			.and_then(|paths| paths.into_iter().next().ok_or(()))
-			.map_err(|_| Bolt12SemanticError::MissingPaths)?;
-		let builder = RefundBuilder::deriving_payer_id(
+		let mut builder = RefundBuilder::deriving_payer_id(
 			node_id, expanded_key, nonce, secp_ctx, amount_msats, payment_id
 		)?
 			.chain_hash($self.chain_hash)
-			.absolute_expiry(absolute_expiry)
-			.path(path);
+			.absolute_expiry(absolute_expiry);
+
+		if let Some(params) = params {
+			let path = $self.create_blinded_paths(params, context)
+				.and_then(|paths| paths.into_iter().next().ok_or(()))
+				.map_err(|_| Bolt12SemanticError::MissingPaths)?;
+
+			builder = builder.path(path);
+		};
```

Comment:
Likewise.

Reply:
Done! [pr3246.06](https://github.com/shaavan/rust-lightning/commits/pr3246.06)
Thanks for the suggestion!